src/components/Filters/Search.js

```diff
@@ -1,21 +1,29 @@
-import { useLocation } from 'react-router-dom';
-
+import { useSearchParams, useLocation } from 'react-router-dom';
 import * as Icons from 'components/Shared/Icons';
 import { useFilter } from 'context/filterContext';
-import { useState, useRef } from 'react';
+import { useState, useEffect, useRef } from 'react';
 import { useDebounce } from 'hooks';
 import { t } from 'i18next';
 import * as Styled from './styles';
 
 const Search = function Search() {
   const { state } = useLocation();
-  const { setSearchValue } = useFilter();
 
+  const { setSearchValue } = useFilter();
+  const [searchParams, setSearchParams] = useSearchParams();
   const defaultValue = state?.searchValue ?? '';
-  const [value, setValue] = useState(defaultValue);
+  const [value, setValue] = useState(searchParams.get('query') || defaultValue);
```

Comment:
@mitjapotocin query is not applied. The reason is while `searchValue` in `filterContext` is not updated because the `value` is immediately set and it's not changing. The reason I have set it to `defaultValue` was that I didn't want to trigger additional re-render when user was coming back from doctor's page (from `<PageInfo>`).

Also we have to use doctor type and availability. Not sure but it looks it would be better to set all filters in `filterContext`  immediately when page loads..



user search
![original](https://user-images.githubusercontent.com/44704999/149502684-06a8b8fb-575b-463c-8066-daceb7de9e29.png)

link
![pasted-log](https://user-images.githubusercontent.com/44704999/149502857-2ee1b698-dcd1-4dd5-b55f-bcf7a2dc1242.png)

Reply:
also you have one old browser error 😄 .